### PR TITLE
Add IoCompat

### PR DESCRIPTION
**Pull request by Darksonn**

### CODE REVIEW

File: src/io.rs [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+        buf: &mut ReadBuf<'_>
+    ) -> Poll<Result<()>> {
+        let me = self.project();
+        let handle = me.handle;
+        let inner = me.inner;
+
+        let unfilled = buf.initialize_unfilled();
+
+        handle.enter(|| {
+            inner.poll_read(cx, unfilled).map_ok(|_num| ())
```
Comment (LucioFranco) [inline src/io.rs]: do we need to advance the buffer here?

File: src/lib.rs [hunk 2]
```diff
@@ -17,7 +17,23 @@ use std::{
 };
 use tokio_02::runtime::{Handle, Runtime};
 
+mod io;
+pub use self::io::IoCompat;
+
 static RT: OnceCell<Runtime> = OnceCell::new();
```
Comment (LucioFranco) [inline src/lib.rs]: nit can we move the static into the `get_handle` method?

File: src/lib.rs [hunk 3]
```diff
@@ -17,7 +17,23 @@ use std::{
 };
 use tokio_02::runtime::{Handle, Runtime};
 
+mod io;
+pub use self::io::IoCompat;
```
Comment (LucioFranco) [inline src/lib.rs:21]: ext trait?

File: src/io.rs (hunk 1 above)
Comment (Darksonn) [reply src/io.rs]: Good catch.

File: src/lib.rs (hunk 3 above)
Comment (Darksonn) [reply src/lib.rs:21]: We would need a lot of ext traits to do that.

File: Cargo.toml [hunk 4]
```diff
@@ -14,8 +14,10 @@ version = "0.1.0"
...
 pin-project-lite = "0.1"
-tokio_02 = {version = "0.2", package = "tokio", features = ["rt-threaded"]}
+tokio_02 = {version = "0.2.22", package = "tokio", features = ["rt-threaded"]}
+tokio_03 = {version = "0.3", package = "tokio"}
+bytes = "0.5.6"
 
 [dev-dependencies]
 hyper = "0.13"
-tokio = {version = "0.3",  features = ["full"]}
+tokio_03 = {version = "0.3", package = "tokio", features = ["full"]}
```
Comment (LucioFranco) [inline Cargo.toml]: The example won't compile will it?

Comment (Darksonn) [reply Cargo.toml]: Ah thanks. I have to do it like this due to the following error:
```
error: the crate `tokio-compat-02 v0.1.0 (/home/alice/src/tokio-compat-02)` depends on crate `tokio v0.3.0` multiple times with different names
```

**Pull request merged**